File: src/main/java/com/github/hongkaiwen/netty/play/http/HttpServerInitiallizer.java
package com.github.hongkaiwen.netty.play.http;

import io.netty.channel.Channel;
import io.netty.channel.ChannelInitializer;
import io.netty.channel.ChannelPipeline;
import io.netty.handler.codec.http.HttpObjectAggregator;
import io.netty.handler.codec.http.HttpServerCodec;

public class HttpServerInitiallizer extends ChannelInitializer {

  @Override
  protected void initChannel(Channel ch) throws Exception {
    ChannelPipeline pipeline = ch.pipeline();
    pipeline.addLast(new HttpServerCodec());
    pipeline.addLast(new HttpObjectAggregator(512*1024));
    pipeline.addLast(new HttpRequestHandler());
  }
}

File: src/main/kotllin/com/github/hongkaiwen/netty/play/kt/http/HttpServerInitiallizer.kt
package com.github.hongkaiwen.netty.play.kt.http

import io.netty.channel.Channel
import io.netty.channel.ChannelInitializer
import io.netty.handler.codec.http.HttpObjectAggregator
import io.netty.handler.codec.http.HttpServerCodec

/**
 * http pipeline 初始化器
 */
class HttpServerInitiallizer : ChannelInitializer<Channel>() {

    override fun initChannel(ch: Channel?) {
        var pipeline = ch?.pipeline()
        pipeline?.addLast(HttpServerCodec())
        pipeline?.addLast(HttpObjectAggregator(1024*1024*64))
        pipeline?.addLast(HttpRequestHandler())
    }

}